src/app/(sidebar)/smart-contracts/contract-explorer/components/VersionHistory.tsx

```diff
@@ -0,0 +1,184 @@
+import { useState } from "react";
+import { Card, Icon, Loader, Text } from "@stellar/design-system";
+
+import { Box } from "@/components/layout/Box";
+import { ErrorText } from "@/components/ErrorText";
+import { useSEContracVersionHistory } from "@/query/external/useSEContracVersionHistory";
+import { formatEpochToDate } from "@/helpers/formatEpochToDate";
+
+import { NetworkType } from "@/types/types";
+
+export const VersionHistory = ({
+  contractId,
+  networkId,
+}: {
+  contractId: string;
+  networkId: NetworkType;
+}) => {
+  type SortDirection = "default" | "asc" | "desc";
+
+  const [sortById, setSortById] = useState("");
+  const [sortByDir, setSortByDir] = useState<SortDirection>("default");
+
+  const {
+    data: versionHistoryData,
+    error: versionHistoryError,
+    isLoading: isVersionHistoryLoading,
+    isFetching: isVersionHistoryFetching,
+  } = useSEContracVersionHistory({
+    networkId,
+    contractId,
+  });
+
+  if (isVersionHistoryLoading || isVersionHistoryFetching) {
+    return (
+      <Box gap="sm" direction="row" justify="center">
+        <Loader />
+      </Box>
+    );
+  }
+
+  if (versionHistoryError) {
+    return (
+      <ErrorText errorMessage={versionHistoryError.toString()} size="sm" />
+    );
+  }
+
+  if (!versionHistoryData) {
+    return (
+      <Text as="div" size="sm">
+        No version history
+      </Text>
+    );
+  }
+
+  type TableHeader = {
```

Comment:
We'll move the `Table` to its own file when implementing another data table. That way, we'll have a better idea of the features we need to support.